editors/sc-ide/widgets/multi_editor.cpp

```diff
@@ -950,8 +948,7 @@ int MultiEditor::insertTab(Document* doc, int insertIndex) {
     tabIdx = mTabs->insertTab(insertIndex, icon, doc->title());
     mTabs->setTabData(tabIdx, QVariant::fromValue<Document*>(doc));
 
-    mDocModifiedSigMap.setMapping(tdoc, doc);
-    connect(tdoc, SIGNAL(modificationChanged(bool)), &mDocModifiedSigMap, SLOT(map()));
+    connect(tdoc, &QTextDocument::modificationChanged, [=](const bool status) { onDocModified(doc); });
```

Comment:
strongly prefer explicit captures for lambdas, and you can simplify the parameter list since you don't use `status` -- `[this, doc](bool) { onDocModified(doc); }`

Reply:
thanks for this - I've brushed up on new(er) Meyers advice and now understand the reasons for this, thanks for the catch!